cmd/codegen/generator/typescript/templates/src/header.ts.gtpl

```diff
@@ -9,6 +9,11 @@ inherited by futures objects and common types.
  */
 import { Context } from "../common/context.js"
 
+/**
+ * Declare a number as float in the Dagger API.
+ */
+export type float = number
```

Comment:
Sigh. This is a real pain that we need this :cry: 

Sadly, I have no better ideas - I think it's the right call to create this distinction in typescript, instead of forcing everything else to deal with a type that is both int+float.

That said, kinda curious what happens if a typescript function returns a `number` and you return a floating point from it - does it just get truncated? Curious what the engine does with it, ideally we should get a nice error message from typescript telling the user to use the custom `float` type instead.

Reply:
Already tried, the engine throws an error because it expect a integer and the `.` is considered as an invalid syntax :)

Reply:
Yeah, having some sort of custom error handling in the typescript SDK would be great :tada:

Reply:
Added: [ceb238b](https://github.com/dagger/dagger/pull/9396/commits/ceb238b167a636244ab48dd738bfc0f32a8255ca)